Bottom: 04f5649bf5734094f8f3e1d4fe0fc9649f7e7bf0
Top:    d50ef3d15c4938a4d95465ec29777931f0df5b20
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2014-07-17 10:10:17 -0400

Refresh of jds-mods

---

diff --git a/sound/soc/sunxi/sunxi-codec.c b/sound/soc/sunxi/sunxi-codec.c
index 5ada5c4..2867159 100644
--- a/sound/soc/sunxi/sunxi-codec.c
+++ b/sound/soc/sunxi/sunxi-codec.c
@@ -59,7 +59,7 @@
 #define MIXEN				(29)
 #define DACPAS				(8)
 #define MIXPAS				(7)
-#define PA_MUTE				(6)
+#define PAMUTE				(6)
 #define PAVOL				(0)
 #define SUNXI_DAC_TUNE		(0x14)
 #define SUNXI_DAC_DEBUG		(0x18)
@@ -124,7 +124,7 @@ static int codec_play_start(struct sunxi_priv *priv)
 
 	/* enable DAC DRQ */
 	regmap_update_bits(priv->regmap, SUNXI_DAC_FIFOC, 0x1 << DAC_DRQ_EN, 0x1 << DAC_DRQ_EN);
-	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x1 << PA_MUTE, 0x1 << PA_MUTE);
+	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x1 << PAMUTE, 0x1 << PAMUTE);
 
 	return 0;
 }
@@ -137,14 +137,12 @@ static int codec_play_stop(struct sunxi_priv *priv)
 #endif
 
 	/* mute PA */
-	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x1 << PA_MUTE, 0x0 << PA_MUTE);
+	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x1 << PAMUTE, 0x0 << PAMUTE);
 	mdelay(5);
 
 	/* disable DAC DRQ */
 	regmap_update_bits(priv->regmap, SUNXI_DAC_FIFOC, 0x1 << DAC_DRQ_EN, 0x0 << DAC_DRQ_EN);
 
-	/* mute PA */
-	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x1 << PA_MUTE, 0x0 << PA_MUTE);
 	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x1 << DACAENL, 0x0 << DACAENL);
 	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x1 << DACAENR, 0x0 << DACAENR);
 
@@ -473,7 +471,7 @@ static int codec_init(struct sunxi_priv *priv)
 		regmap_update_bits(priv->regmap, SUNXI_DAC_DPC, 0x3F << DVOL, 0 << DVOL);
 
 	/* PA mute */
-	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 1 << PA_MUTE, 0 << PA_MUTE);
+	regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 1 << PAMUTE, 0 << PAMUTE);
 
 	/* enable PA */
 	regmap_update_bits(priv->regmap, SUNXI_ADC_ACTL, 1 << PA_EN, 1 << PA_EN);
@@ -483,7 +481,7 @@ static int codec_init(struct sunxi_priv *priv)
 	if (priv->id == SUN4A)
 		regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x3f << PAVOL, 1 << PAVOL);
 	else
-		regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x3f << PAVOL, 0x3b << PAVOL);
+		regmap_update_bits(priv->regmap, SUNXI_DAC_ACTL, 0x3f << PAVOL, 0x30 << PAVOL);
 
 	return 0;
 }
